tools/donate_cpu_lib.py

```diff
@@ -42,12 +42,19 @@ def get_cppcheck(cppcheck_path, work_path):
             try:
                 os.chdir(cppcheck_path)
                 try:
+                    subprocess.check_call(['git', 'fetch'])
+                    localrev = subprocess.check_output(['git', 'rev-parse', 'main'])
+                    origrev = subprocess.check_output(['git', 'rev-parse', 'origin/main'])
+                    if localrev == origrev:
+                        print('Already up to date. Continuing with %s.' % localrev.decode().rstrip())
+                        time.sleep(2)
+                        return True
                     subprocess.check_call(['git', 'checkout', '-f', 'main'])
                 except subprocess.CalledProcessError:
```

Comment:
On a side note - I think we can get rid of this fallback. The branch was renamed quite a while ago.

Reply:
I don't think the fallback is necessary either, but I think we could do (untested):
```
git fetch origin
git checkout -f -B main origin/main
```
and remove the fallback, which *should* work regardless of how old the repo is, if there are local changes (commited or uncommited), even a force-push to the cppcheck repo *should* be handled.

Reply:
Okay. I will test this, then update the pull request.